src/plugins/auto_batch/src/plugin.cpp

```diff
@@ -49,311 +46,359 @@ DeviceInformation Plugin::ParseBatchDevice(const std::string& deviceWithBatch) {
-                                           batched_outputs);
-}
-
-InferenceEngine::IExecutableNetworkInternal::Ptr Plugin::LoadExeNetworkImpl(
-    const InferenceEngine::CNNNetwork& network,
-    const std::shared_ptr<InferenceEngine::RemoteContext>& context,
-    const std::map<std::string, std::string>& user_config) {
-    return LoadNetworkImpl(network, context, user_config);
+                                           batched_outputs,
+                                           compiled_model_with_batch,
+                                           compiled_model_without_batch);
 }
 
-InferenceEngine::QueryNetworkResult Plugin::QueryNetwork(const InferenceEngine::CNNNetwork& network,
-                                                         const std::map<std::string, std::string>& user_config) const {
-    auto core = GetCore();
-    if (!core)
-        return InferenceEngine::QueryNetworkResult();
-    auto cfg = user_config;
+ov::SupportedOpsMap Plugin::query_model(const std::shared_ptr<const ov::Model>& model,
+                                        const ov::AnyMap& properties) const {
+    OPENVINO_ASSERT(model, "OpenVINO Model is empty!");
+    OPENVINO_ASSERT(get_core(), "Core is missing!");
+    auto cfg = properties;
     for (const auto& c : cfg) {
+        OPENVINO_SUPPRESS_DEPRECATED_START
         if (c.first == CONFIG_KEY(AUTO_BATCH_DEVICE_CONFIG) || c.first == ov::device::priorities.name()) {
             auto val = c.second;
             cfg.erase(c.first);
-            auto metaDevice = ParseMetaDevice(val, cfg);
-            return core->QueryNetwork(network, metaDevice.device_name, cfg);
+            auto metaDevice = parse_meta_device(val.as<std::string>(), cfg);
+            return get_core()->query_model(model, metaDevice.device_name, cfg);
         }
+        OPENVINO_SUPPRESS_DEPRECATED_END
     }
-    IE_THROW() << "Value for KEY_AUTO_BATCH_DEVICE_CONFIG is not set";
+    OPENVINO_THROW("Value for MULTI_DEVICE_PRIORITIES for AUTO BATCH PLUGIN is not set");
+}
+
+std::shared_ptr<ov::IRemoteContext> Plugin::get_default_context(const ov::AnyMap& remote_properties) const {
+    auto full_properties = merge_properties(m_plugin_config, remote_properties);
+    OPENVINO_SUPPRESS_DEPRECATED_START
+    auto device_batch = full_properties.find(CONFIG_KEY(AUTO_BATCH_DEVICE_CONFIG));
+    if (device_batch == full_properties.end())
+        device_batch = full_properties.find(ov::device::priorities.name());
+    if (device_batch == full_properties.end()) {
+        OPENVINO_THROW("MULTI_DEVICE_PRIORITIES key for AUTO NATCH is not set for BATCH device");
+    }
+    OPENVINO_SUPPRESS_DEPRECATED_END
+
+    auto val = device_batch->second.as<std::string>();
+    auto core = get_core();
+    if (!core)
+        return nullptr;
+    auto metaDevice = parse_meta_device(val, ov::AnyMap());
+    auto remote_context = core->get_default_context(metaDevice.device_name);
+    return remote_context._impl;
```

Comment:
the same - can we create wrapper on top of HW-remote context ?

Reply:
seems applicable, here we can implement the wrapper